query-sql/src/lib.rs

```diff
@@ -127,6 +136,35 @@ fn push_column(qb: &mut QueryBuilder, col: &DatomsColumn) {
 //---------------------------------------------------------
 // Turn that representation into SQL.
 
+
+/// A helper macro to sequentially process an iterable sequence,
+/// evaluating a block between each pair of items.
+///
+/// This is used to simply and efficiently produce output like
+///
+/// ```sql
+///   1, 2, 3
+/// ```
+///
+/// or
+///
+/// ```sql
+/// x = 1 AND y = 2
+/// ```
+///
+/// without producing an intermediate string sequence.
```

Comment:
You don't need to do this now, but I think you'd find a plain function taking closures easier to read: there's no "dangling ident", and each closure can take `|out|` or whatever.